File: advent/2019/6/part1.cpp
#include<iostream>
#include<boost/algorithm/string.hpp>
#include<string>
#include<map>
#include<vector>

using namespace std;

int main(){
    //map<string, vector<string> > orbits;
    //map<string, int> num_orbits;
    
    string planet1;
    string planet2;
    string line;
    while(cin>> line){
      vector<string> planets;
      boost::split(planets, line, [](char c){return c == ')';});
      planet1 = planets[0];
      planet2 = planets[1];
      cout << planet1 << " "<< planet2 << endl;
      //if(orbits.count(planet1) == 1) orbits[planet1].push_back(planet2);
      //else orbits[planet1] = vector<string>{planet2};
    }

}

File: advent/2019/6/part1.py
from sys import stdin

orbits = {}
planet_list = []
orbits_num = {}

def rec(planet):
    if(planet in orbits_num): return orbits_num[planet]
    if(not planet in orbits): return 0
    else: return 1 + rec(orbits[planet])

for line in stdin:
    planet1, planet2 = line.strip().split(')')
    orbits[planet2] = planet1
    if(not planet1 in planet_list): planet_list.append(planet1)
    if(not planet2 in planet_list): planet_list.append(planet2)

count = 0
for planet in planet_list:
    count += rec(planet)

print(count)
